File: back/central-reserve/internal/infra/primary/http2/handlers/authhandler/response/login-response.go
package response

import "time"

// LoginResponse representa la respuesta simplificada del login
type LoginResponse struct {
	User                  UserInfo       `json:"user"`
	Token                 string         `json:"token"`
	RequirePasswordChange bool           `json:"require_password_change"`
	Businesses            []BusinessInfo `json:"businesses"`
}

// UserInfo representa la información del usuario en la respuesta
type UserInfo struct {
	ID          uint       `json:"id"`
	Name        string     `json:"name"`
	Email       string     `json:"email"`
	Phone       string     `json:"phone"`
	AvatarURL   string     `json:"avatar_url"`
	IsActive    bool       `json:"is_active"`
	LastLoginAt *time.Time `json:"last_login_at"`
}

// RoleInfo representa la información del rol en la respuesta
type RoleInfo struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	Code        string `json:"code"`
	Description string `json:"description"`
	Level       int    `json:"level"`
	Scope       string `json:"scope"`
}

// PermissionInfo representa la información del permiso en la respuesta
type PermissionInfo struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	Code        string `json:"code"`
	Description string `json:"description"`
	Resource    string `json:"resource"`
	Action      string `json:"action"`
	Scope       string `json:"scope"`
}

// ResourcePermissions representa los permisos agrupados por recurso
type ResourcePermissions struct {
	Resource     string           `json:"resource"`
	ResourceName string           `json:"resource_name"`
	Actions      []PermissionInfo `json:"actions"`
}

// UserRolesPermissionsResponse representa la respuesta de roles y permisos del usuario
type UserRolesPermissionsResponse struct {
	IsSuper   bool                  `json:"is_super"`
	Roles     []RoleInfo            `json:"roles"`
	Resources []ResourcePermissions `json:"resources"`
}

// LoginSuccessResponse representa la respuesta exitosa del login para Swagger
type LoginSuccessResponse struct {
	Success bool          `json:"success"`
	Data    LoginResponse `json:"data"`
}

// UserRolesPermissionsSuccessResponse representa la respuesta exitosa de roles y permisos
type UserRolesPermissionsSuccessResponse struct {
	Success bool                         `json:"success"`
	Data    UserRolesPermissionsResponse `json:"data"`
}

// LoginErrorResponse representa la respuesta de error del login para Swagger
type LoginErrorResponse struct {
	Error   string `json:"error"`
	Details string `json:"details,omitempty"`
}

// LoginBadRequestResponse representa la respuesta de error 400 para Swagger
type LoginBadRequestResponse struct {
	Error   string `json:"error"`
	Details string `json:"details"`
}

// BusinessInfo representa la información del negocio en la respuesta
type BusinessInfo struct {
	ID                 uint             `json:"id"`
	Name               string           `json:"name"`
	Code               string           `json:"code"`
	BusinessTypeID     uint             `json:"business_type_id"`
	BusinessType       BusinessTypeInfo `json:"business_type"`
	Timezone           string           `json:"timezone"`
	Address            string           `json:"address"`
	Description        string           `json:"description"`
	LogoURL            string           `json:"logo_url"`
	PrimaryColor       string           `json:"primary_color"`
	SecondaryColor     string           `json:"secondary_color"`
	TertiaryColor      string           `json:"tertiary_color"`
	QuaternaryColor    string           `json:"quaternary_color"`
	NavbarImageURL     string           `json:"navbar_image_url"`
	CustomDomain       string           `json:"custom_domain"`
	IsActive           bool             `json:"is_active"`
	EnableDelivery     bool             `json:"enable_delivery"`
	EnablePickup       bool             `json:"enable_pickup"`
	EnableReservations bool             `json:"enable_reservations"`
}

// BusinessTypeInfo representa la información del tipo de negocio
type BusinessTypeInfo struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	Code        string `json:"code"`
	Description string `json:"description"`
	Icon        string `json:"icon"`
}

File: front/rupu-central/src/services/auth/login/infrastructure/repositories/response/login.response.ts
/**
 * Interfaces de respuesta del backend para Login
 */

export interface BackendUserInfo {
  id: number;
  name: string;
  email: string;
  phone?: string;
  avatar_url?: string;
  is_active: boolean;
  last_login_at?: string | null;
}

export interface BackendBusinessTypeInfo {
  id: number;
  name: string;
  code: string;
  description?: string;
  icon?: string;
}

export interface BackendBusinessInfo {
  id: number;
  name: string;
  code: string;
  business_type_id: number;
  business_type: BackendBusinessTypeInfo;
  timezone?: string;
  address?: string;
  description?: string;
  logo_url?: string;
  primary_color?: string;
  secondary_color?: string;
  tertiary_color?: string;
  quaternary_color?: string;
  navbar_image_url?: string;
  custom_domain?: string;
  is_active: boolean;
  enable_delivery?: boolean;
  enable_pickup?: boolean;
  enable_reservations?: boolean;
}

export interface BackendLoginResponseData {
  user: BackendUserInfo;
  token: string;
  require_password_change: boolean;
  businesses: BackendBusinessInfo[];
  scope: string;
  is_super_admin: boolean;
}

export interface BackendLoginResponse {
  success: boolean;
  data: BackendLoginResponseData;
}

export interface BackendLoginErrorResponse {
  error: string;
  details?: string;
}
